Dikubot/Webapp/Data/Emojis/EmojiService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Dikubot.Extensions.EmojiSelector;
+using Newtonsoft.Json;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using Dikubot.Webapp.Shared.Emojis;
+using Microsoft.AspNetCore.Components;
+using Dikubot.Webapp.Data.Emojis;
+using Syncfusion.Lic.util.encoders;
+
+namespace Dikubot.Webapp.Data
+{
+    public class EmojiService
+    {
+        
+        public List<EmojiGroup> EmojiLists;
+        private Dictionary<string, string> _twitterEmojis;
+        
+        public string emptyEmoji = @"<svg xmlns=""http://www.w3.org/2000/svg"" viewBox=""0 0 36 36""></svg>";
+        
+
+        public EmojiService()
+        {
+            string EmojisString = File.ReadAllText("Webapp/wwwroot/misc/emoji-all-groups.json");
+
+            _twitterEmojis = typeof(Twemoji).GetFields()
+                .ToDictionary(m =>
+                {
+                    var res = m.Name
+                        .Replace("E", "").Replace("_", " ")
+                        .ToUpper();
+                    
+                    return res;
+                }, info => info.GetValue(info).ToString());
+
+            EmojiLists = JsonConvert.DeserializeObject<List<EmojiGroup>>(EmojisString);
+
+            EmojiLists = EmojiLists.Select(emojiGroup =>
+            {
+                emojiGroup.emojiList.Select(emojiList =>
+                {
+                    emojiList.svg = GetSvgValue(emojiList.unicode);
+                    if (emojiList.skins != null)
+                    {
+                        emojiList.skins.Insert(0,new Skin(){svg = emojiList.svg, unicode = emojiList.unicode});
+                        emojiList.skins = emojiList.skins.Select(skin =>
+                        {
+                            skin.svg = GetSvgValue(skin.unicode);
+                            return skin;
+                        }).Where(emoji => emoji.svg != emptyEmoji).ToList();
+                    }
+                    return emojiList;
+                }).Where(emoji => emoji.svg != emptyEmoji).ToList();
+                return emojiGroup;
+            }).ToList();
```

Comment:
This code is not very nice to look at. Perhaps consider how you can simplify it, and split it into multiple methods, to make it prettier and easier to read.

I found a quick article about how to avoid nesting:
https://javascript.plainenglish.io/the-secret-to-clean-and-efficient-code-how-to-denest-your-code-a00cac3449e6